### PR TITLE
Connected to firebase, fixed create channel model

### PR DESCRIPTION
**Pull request by angelica-hernandez**
Some edits from Shi Bin as well

### CODE REVIEW

File: src/app/chatbox/chatbox.component.html [hunk 1]
```diff
@@ -19,15 +19,15 @@
...
+            <mat-list-item *ngIf="roomname">
+              <b>{{roomname}}</b>
             </mat-list-item>
           </mat-list>
         </mat-tab>
       </mat-tab-group>
       <mat-menu #appMenu="matMenu" yPosition="above">
         <button mat-menu-item (click)="openDialog()">Create Channel</button>
-        <button mat-menu-item>Private Message Individual</button>
+        <button mat-menu-item (click)="sendToFirebase()">Private Message Individual</button>
```
Comment (shiBinH) [inline src/app/chatbox/chatbox.component.html]: You can remove the function call on clicks. I unintentionally added this before.

**Comment by Hongzred** [comment]
checked the code locally and also checked the firebase, it does create a new chatroom to our database. There still need some work on the ui data binding. But I think after adding the fixes @shiBinH mentioned above, this can be merged into master. But before that, it will need to rebase the master after #93 gets merged into master.

File: src/app/chatbox/chatbox.component.html [hunk 2]
```diff
@@ -19,15 +19,15 @@
                 >delete</mat-icon
               >
             </mat-list-item> -->
-            <mat-list-item *ngIf="chatroomName">
-              <b>{{chatroomName}}</b>
+            <mat-list-item *ngIf="roomname">
+              <b>{{roomname}}</b>
```
[the remark this answers is not in the document]
Comment (angelica-hernandez) [reply src/app/chatbox/chatbox.component.html]: all instances of roomname have been changed to roomName now

File: src/app/chatbox/chatbox.component.ts [hunk 3]
```diff
@@ -31,6 +31,10 @@ export class ChatboxComponent implements OnInit {
   messages: string[] = [];
   secretCode = 'secret';
   friendListId = [];
+  status: string;
+  roomname: string;
+  userList: string[];
+  ownerID: string;
```
[the remark this answers is not in the document]
Comment (angelica-hernandez) [reply src/app/chatbox/chatbox.component.ts]: removed all except roomName

File: src/app/createchannel/createchannel.component.html [hunk 4]
```diff
@@ -6,13 +6,13 @@ <h1 mat-dialog-title>Say hello...</h1>
...
+          <input matInput [(ngModel)]="roomname">
         </mat-form-field>
       </div>
       </mat-card>
   </mat-nav-list>
 </div>
 <div mat-dialog-actions>
   <button mat-button (click)="onNoClick()">Exit</button>
-  <button mat-button [mat-dialog-close]="data.chatroomName" >Start Chatting!</button>
+  <button mat-button (click)="sendToFirebase()" [mat-dialog-close]="roomname" >Start Chatting!</button>
```
[the remark this answers is not in the document]
Comment (angelica-hernandez) [reply src/app/createchannel/createchannel.component.html]: I still need the selector [mat-dialog-close] to actually close out of the dialog, so, for now, I put empty quotes.

File: src/app/chatbox/chatbox.component.ts [hunk 5]
```diff
@@ -168,21 +172,27 @@ export class ChatboxComponent implements OnInit {
...
+      }
     });
 
     dialogRef.afterClosed().subscribe(result => {
       console.log('The dialog was closed');
-      this.chatroomName = result;
+      this.roomname = result;
+      console.log(result);
     });
   }
```
[the remark this answers is not in the document]
Comment (angelica-hernandez) [reply src/app/chatbox/chatbox.component.ts]: removed

**Pull request merged**